frontend/public/components/environment.jsx

```diff
@@ -220,15 +327,32 @@ export const EnvironmentPage = connect(stateToProps)(
     static getDerivedStateFromProps(nextProps, prevState) {
       const { currentEnvVars } = prevState;
       const { rawEnvData, readOnly } = nextProps;
-      const incomingEnvVars = envVarsToArray(rawEnvData);
-      if (_.isEqual(incomingEnvVars, currentEnvVars)) {
+      const incomingEnvVars = new CurrentEnvVars();
+      incomingEnvVars.instantiateWithRawData(rawEnvData);
+      if (currentEnvVars.isEqual(incomingEnvVars)) {
         return null;
       }
-      return readOnly ? {currentEnvVars: incomingEnvVars} : {stale: true, success: null};
+      return readOnly ? {
+        currentEnvVars
+      } : {stale: true, success: null};
     }
 
-    _selectContainer(index) {
-      this.setState({containerKey: parseInt(index, 10)});
+    _selectContainer(containerName) {
+      const { rawEnvData } = this.props;
+      let containerKey = _.findIndex(rawEnvData.containers, {name: containerName});
+      if (containerKey >= 0) {
```

Comment:
Is it a key or an index? This state value seems misnamed

Reply:
sure, can change the name to containerIndex